### PR TITLE
Using the Unity 2019.3 Device Simulator Format

### PR DESCRIPTION
**Pull request by 06Games**
- [x] Get simulation data from JSON files
- [x] Convert current data to the new format
- [x] Convert mock-up to the Unity system
- [x] Custom path for data collection
- [x] Extract data directly from the debug scene

### CODE REVIEW

**Comment by 5argon** [comment]
Thank you! I have a massive refactor pushed that may impact your PR however. Some `NotchSimulatorUtility` was changed to `NotchSimulatorUtilityEditor` for instance.

**Comment by 06Games** [comment]
Ok, I will look at your changes and I will update this PR as soon as possible to comply with them.

**Comment by 06Games** [comment]
I would like your opinion on the implementation of the overlay system.
The Unity simulator requires only one overlay per device, but does not take into account visual elements such as the lower bar on iOS.
So, I'd like to know if I should keep the principle of one overlay per rotation or if I comply with Unity and lose the display of the iOS bar.

File: DebugScene/NotchSolutionDebugger.cs [hunk 1]
```diff
@@ -70,20 +99,80 @@ Rect RealSize(Rect relative)
...
+            }
+        }
+
+        device.Screens = new ScreenData[1];
+        for(int i = 0; i < device.Screens.Length; i++)
+        {
+            var screen = new ScreenData();
+            screen.width = Screen.width;
+            screen.height = Screen.height;
+            //screen.navigationBarHeight = 0;
```
Comment (06Games) [inline DebugScene/NotchSolutionDebugger.cs:141]: I don't know how to get this value. If you have an idea...

**Comment by 5argon** [comment]
> I would like your opinion on the implementation of the overlay system.
> The Unity simulator requires only one overlay per device, but does not take into account visual elements such as the lower bar on iOS.
> So, I'd like to know if I should keep the principle of one overlay per rotation or if I comply with Unity and lose the display of the iOS bar.

I think just one is fine, likely the definition of overlay of Unity did not include on-screen elements and just the frame and notches.

You did a lot of work! Much appreciated! But I kinda feel bad that my very rough plan ended up as your work and its done almost immediately..

I had been busy releasing an update in a remake of my old game I worked for ~6 months but it seems to be settled now. (Also is a dogfooding of Notch Solution in production, the game is called Duel Otters and it adapts to notches.) I will get back to look at this soon + add support for cutouts soon to use in this game also. There are some corner elements I would like it to stay on iPhone but moves down on S10+.

Also I just watched https://www.youtube.com/watch?v=22HIEQTyozQ and planning to make this a proper UPM shape soon.

**Comment by 06Games** [comment]
Okay, so I will improve the implementation I made and this pull request will be finished.
You did not have to apologize, I understand that you could devote more investment. That's why I'm working on improving this project. It's the strength of open-source community participation, not everyone has the time and means to finish alone.
I'm glad you have some more time dedicated to the development of NotchSolution.
In addition, I think it would be interesting to use the Github versioning system. It would be a quick way to follow the evolution of the project as and when updates and to find the version that interests us.

**Comment by 06Games** [comment]
I think this pull request is now ready to be merged. You can now review it.

**Pull request merged**